### PR TITLE
Allow customization of build tool tasks

### PR DESCRIPTION
**Pull request by anton-matosov**
Implements #305

Add args parameter to all build tool tasks that allows customization in the workspaces's tasks.json by users
Implement generic task named `ros`
Add `roscore` and `roslaunch` commands using `ros` task that allows user to run and customize them in tasks.json

### CODE REVIEW

**Comment by seanyen** [comment]
Thank @anton-matosov! Can you rebase to reduce the noise on the diff view?

**Comment by anton-matosov** [comment]
Definitely. my bad

File: .vscode/settings.json [hunk 1]
```diff
@@ -11,6 +11,5 @@
     "search.exclude": {
         "out": true // set this to false to include "out" folder in search results
     },
-    "typescript.tsdk": "./node_modules/typescript/lib", // we want to use the TS server from our node_modules folder to control its version
-    "python.pythonPath": "C:\\opt\\python27amd64\\python.exe"
+    "typescript.tsdk": "./node_modules/typescript/lib"
```
Comment (anton-matosov) [inline .vscode/settings.json:14]: This is the automatic upgrade of settings by corresponding vscode extensions

**Comment by seanyen** [comment]
@anton-matosov, LGTM. Is it ready for merge too?

**Comment by ooeygui** [comment]
I'm testing this on Windows. @anton-matosov can you update your branch so that the PR is in sync?

File: doc/debug-support.md [hunk 2]
```diff
@@ -41,6 +41,96 @@ The launch-debug flow provided by `vscode-ros` will not spawn a `rosmaster`.
 
 ![launch and debug Python and C++ nodes][launch_and_debug_nodes]
 
+### <a name="build_tasks"></a> Use tasks to automatically build before starting debug session
+
+The first thing you need to do is to create build task for your package(s) with enabled debug symbols. 
+In the example below you can see a `catkin_make` build task that passes additional `-DCMAKE_BUILD_TYPE=Debug` argument that switches build to use `Debug` configuration, which is the most suitable configuration for debugging, because it has 0 optimization level and includes debug symbols.
```
[the remark this answers is not in the document]
Comment (anton-matosov) [reply doc/debug-support.md:47]: On macOS I found that `RelWithDebInfo` has optimizations on and many local variables are inlined and stepping misbehaves.
For windows builds I would assume primary bottleneck is debug version of CRT and thus you need to compile Release. 
So we might want to talk about both options and corresponding pros and cons.

**Comment by anton-matosov** [comment]
Merged in latest master and updated docs

**Pull request merged**